packages/runtime/src/router.ts

```diff
@@ -487,6 +487,9 @@ export const createRouter = <E extends Env = Env, S extends Schema = Schema>():
 			const asyncLocalStorage = getAgentAsyncLocalStorage();
 			const capturedContext = asyncLocalStorage.getStore();
 
+			// Set Content-Type header for streaming response detection by clients
+			c.header('Content-Type', 'application/octet-stream');
```

Comment:
good enough to unblock this but we probably want to follow-up and look at the schema to determine the content-type too.